src/Symfony/Component/OptionsResolver/OptionsResolver.php

```diff
@@ -956,6 +936,24 @@ public function count()
     }
 
     /**
+     * Helper method, which returns UndefinedOptionsException.
+     * Returned exception can be thrown from methods, which
+     * require option to be set, such like setNormalizer(), addAllowedTypes() etc.
+     *
+     * @param string $option The undefined option name, which will be used in exception message
+     *
+     * @return UndefinedOptionsException
+     */
+    private function createUndefinedOptionException($option)
+    {
+        return new UndefinedOptionsException(sprintf(
+            'The option "%s" does not exist. Defined options are: "%s".',
+            $option,
+            implode('", "', array_keys($this->defined))
```

Comment:
Why not simply create a dedicated exception and throw that? You could move this logic to the constructor.